cli/pkg/kctrl/cmd/app/init/init.go

```diff
@@ -0,0 +1,336 @@
+	//TODO what needs to be added here
+}
+
+func (createStep *CreateStep) PreInteract() error {
+	createStep.printStartBlock()
+	return nil
+}
+
+func (createStep *CreateStep) Interact() error {
+	err := createStep.configureFetchSection()
+	if err != nil {
+		return err
+	}
+
+	err = createStep.configureTemplateSection()
+	if err != nil {
+		return err
+	}
+
+	createStep.configureExportSection()
+
+	return nil
+}
+
+func (createStep *CreateStep) configureFetchSection() error {
+	fetchConfiguration := fetch.NewFetchStep(createStep.ui, createStep.appBuild, createStep.isAppCommandRunExplicitly)
+	err := common.Run(fetchConfiguration)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep *CreateStep) configureTemplateSection() error {
+	templateConfiguration := template.NewTemplateStep(createStep.ui, createStep.appBuild)
+	err := common.Run(templateConfiguration)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep *CreateStep) PostInteract() error {
+	if createStep.isAppCommandRunExplicitly {
+		appConfig, err := createStep.generateApp()
+		if err != nil {
+			return err
+		}
+
+		appContent, err := yaml.Marshal(appConfig)
+		if err != nil {
+			return err
+		}
+		err = common.WriteFile(AppFileName, appContent)
+		if err != nil {
+			return err
+		}
+
+		createStep.ui.PrintInformationalText(fmt.Sprintf("Both the files can be accessed from the following location:"))
+	}
+
+	return nil
+}
+
+func (createStep CreateStep) generateApp() (kcv1alpha1.App, error) {
+	var app kcv1alpha1.App
+	exists, err := common.IsFileExists(AppFileName)
+	if err != nil {
+		return kcv1alpha1.App{}, err
+	}
+
+	if exists {
+		app, err = createStep.updateExistingApp()
+		if err != nil {
+			return kcv1alpha1.App{}, err
+		}
+	} else {
+		app = createStep.createAppFromAppBuild()
+	}
+	timestamp := v1.NewTime(time.Now().UTC()).Rfc3339Copy()
+	app.ObjectMeta.CreationTimestamp = timestamp
+	return app, nil
+
+}
+
+func printNextStep(ui cmdcore.AuthoringUI) {
+	ui.PrintInformationalText("\n**Next steps**")
+	ui.PrintInformationalText("\nCreated app can be consumed in following ways:\n")
+}
+
+func printInformation(ui cmdcore.AuthoringUI) {
+	ui.PrintInformationalText("\n**Information**\napp-build.yml is generated as part of this flow. This file can be used for further updating and adding complex scenarios while using the `kctrl dev deploy` command. Please read the link'ed documentation for more explanation.")
+}
+
+func (createStep CreateStep) createAppFromAppBuild() kcv1alpha1.App {
+	//TODO Should we ask for the app Name ?
+	appName := "microservices-demo"
+	serviceAccountName := fmt.Sprintf("sa-%s", appName)
+	appAnnotation := map[string]string{
+		fetch.LocalFetchAnnotationKey: ".",
+	}
+	appTemplateSection := createStep.appBuild.Spec.App.Spec.Template
+	//TODO should we remove the fetch section as it is not beind used and add it dynamically during dev deploy.
+	appFetchSection := []kcv1alpha1.AppFetch{kcv1alpha1.AppFetch{
```

Comment:
I am not adding the fetchSection to generated AppBuild as this is not being used. Should we add it?